search/lib/api/stac.js

```diff
@@ -20,10 +21,10 @@ async function getSearch (request, response) {
   const query = stacExtension.stripStacExtensionsFromRequestObject(request.query); // The cmr function to convert params can not handle stac extensions
   const params = Object.assign({ provider: providerId }, query);
   const convertedParams = cmr.convertParams(cmr.STAC_QUERY_PARAMS_CONVERSION_MAP, params);
-  const result = await search(event, convertedParams);
-  await assertValid(schemas.items, result);
-  const formatedResult = stacExtension.applyStacExtensions(request.query, result); // Apply any stac extensions that are present
-  response.status(200).json(formatedResult);
+  const { searchResult, featureCollection } = await search(event, convertedParams);
+  await assertValid(schemas.items, featureCollection);
+  const formatted = stacExtension.applyStacExtensions(featureCollection, { fields: request.params.fields, context: { searchResult, query } }); // Apply any stac extensions that are present
```

Comment:
Changed API of applyStacExtensions a bit, caller has to know a little bit more about the extensions, but I think it's slightly better to have it organized in this object than passing in a bunch of random stuff. But let me know what you think.